### PR TITLE
Add UTF8 support for identifiers in Vertica dialect

### PR DESCRIPTION
**Pull request by troshnev**
> https://docs.vertica.com/24.3.x/en/sql-reference/language-elements/identifiers/ :
> Unquoted SQL identifiers must begin with one of the following:
> * Non-Unicode letters: A–Z or a-z
> -- /actually Vertica accepts also non-ASCII UTF-8 Unicode characters here, which is not well documented/
> * Underscore (_)
> Subsequent characters in an identifier can be any combination of the following:
> * Non-Unicode letters: A–Z or a-z
> * Underscore (_)
> * Digits(0–9)
> * Unicode letters (letters with diacriticals or not in the Latin alphabet), unsupported for model names
> * Dollar sign ($), unsupported for model names
>
> Vertica accepts **non-ASCII UTF-8 Unicode characters** for table names, column names, and other identifiers,
> extending the cases where upper/lower case distinctions are ignored (case-folded) to all alphabets,
> including Latin, Cyrillic, and Greek.
>
> Changes to be committed:
>	modified:   src/sqlfluff/dialects/dialect_vertica.py
>	new file:   test/fixtures/dialects/vertica/utf8.sql

### Pull Request checklist
- [ ] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: test/fixtures/dialects/vertica/utf8.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+--https://docs.vertica.com/24.3.x/en/sql-reference/language-elements/identifiers/
```
Comment (WittierDinosaur) [inline test/fixtures/dialects/vertica/utf8.sql:1]: This is quite a fundamental change, looks good though. Woudl you mind adding more test cases?

**Comment by troshnev** [comment]
Fixed all known issues at the moment, ready for next review

**Comment by skryzh** [comment]
@WittierDinosaur hi, is it possible to change my incorrect login in commit to correct one?
sergey.kry -> skryzh
It was a rookie mistake with local git config file...

**Comment by alanmcruickshank** [comment]
> @WittierDinosaur hi, is it possible to change my incorrect login in commit to correct one? sergey.kry -> skryzh It was a rookie mistake with local git config file...

I'll let @WittierDinosaur re-review this PR given he did the first pass, however on changing the commit username: @skryzh I think you can do that on your side. The branch for this PR is in the `troshnev` fork and not in the main repository, so you've still got full control if you want to change the commit username. It might be easiest to do this by doing a soft-reset on this branch (i.e. uncommitting your changes but keeping them) and then re-committing them using your correct username. You'd then need to force-push to the remote branch used for this PR. As the admin for your fork, you're best placed to make that change rather than us.

**Review by alanmcruickshank — APPROVED** [review]
Looks very sensible. Thanks for contributing this and putting together the extensive test case 👍

**Pull request merged**